File: python/svix/models/endpoint_upsert_in.py
# this file is @generated
import typing as t

from .common import BaseModel


class EndpointUpsertIn(BaseModel):
    description: t.Optional[str] = None

    throttle_rate: t.Optional[int] = None
    """Maximum messages per second to send to this endpoint.

    Outgoing messages will be throttled to this rate."""

    uid: t.Optional[str] = None
    """Optional unique identifier for the endpoint."""

    url: str

    disabled: t.Optional[bool] = None

    filter_types: t.Optional[t.List[str]] = None

    channels: t.Optional[t.List[str]] = None
    """List of message channels this endpoint listens to (omit for all)."""

    metadata: t.Optional[t.Dict[str, str]] = None

File: kotlin/lib/src/main/kotlin/models/EndpointUpsertIn.kt
// This file is @generated
package com.svix.kotlin.models

import kotlinx.serialization.Serializable

@Serializable
data class EndpointUpsertIn(
    val url: String,
    val description: String? = null,
    /**
     * Maximum messages per second to send to this endpoint.
     *
     * Outgoing messages will be throttled to this rate.
     */
    val throttleRate: UShort? = null,
    /** Optional unique identifier for the endpoint. */
    val uid: String? = null,
    val disabled: Boolean? = null,
    val eventTypes: Set<String>? = null,
    /** List of message channels this endpoint listens to (omit for all). */
    val channels: Set<String>? = null,
    val metadata: Map<String, String>? = null,
)
